oauth_provider/README.rst

```diff
@@ -0,0 +1,147 @@
+   - Backend Application : User Credentials Grant (not implemented yet)
+
+ - The skip authorization checkbox allows the client to skip the authorization page, and directly deliver a token without prompting the user (useful when the application is trusted).
+ - The allowed scopes list defines which data will be accessible by this client applicaton.
+ - The allowed redirect URIs must match the URI sent by the client, to avoid redirecting users to an unauthorized service. The first value in the list is the default redirect URI.
+
+For example, to configure an Odoo's *auth_oauth* module compatible client, you will enter these values :
+
+- Name : Anything you want
+- Client identifier : The identifier you want to give to this client
+- Application Type : Mobile Application (Odoo uses the implicit grant mode, which corresponds to the mobile application profile)
+- Allowed Scopes : Nothing required, but allowing access to current user's email and name is used by Odoo to fill user's information on signup
+- Allowed Redirect URIs : http://odoo.example.com/auth_oauth/signin
+
+Usage
+=====
+
+Server
+------
+
+This module will allow OAuth clients to use your Odoo instance as an OAuth provider.
+
+Once configured, you must give these information to your client application :
+
+#. Client identifier : Identifies the application (to be able to check allowed scopes and redirect URIs)
+#. Allowed scopes : The codes of scopes allowed for this client
+#. URLs for the requests :
+
+  - Authorization request : http://odoo.example.com/oauth2/authorize
+  - Token request : http://odoo.example.com/oauth2/token
+  - Token information request : http://odoo.example.com/oauth2/tokeninfo
+     Parameters : access_token
+  - User information request : http://odoo.example.com/oauth2/userinfo
+     Parameters : access_token
+  - Any other model information request (depending on the scopes) : http://odoo.example.com/oauth2/data
+     Parameters : access_token and model
+
+For example, to configure the *auth_oauth* Odoo module as a client, you will enter these values :
+
+- Provider name : Anything you want
+- Client ID : The identifier of the client configured in your Odoo Provider instance
+- Body : Text displayed on Odoo's login page link
+- Authentication URL : http://odoo.example.com/oauth2/authorize
+- Scope : A space separated list of scope codes allowed to the client in your Odoo Provider instance
+- Validation URL : http://odoo.example.com/oauth2/tokeninfo
+- Data URL : http://odoo.example.com/oauth2/userinfo
+
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/149/10.0
+
+Known issues / Roadmap
+======================
+
+* Implement the backend application profile (client credentials grant type)
+* Add checkboxes on the authorization page to allow the user to disable some scopes for a token ? (I don't know if this is allowed in the OAuth protocol)
```

Comment:
Oh cool yes, this is totally allowed in OAuth spec. I think though that this is actually implemented client side. Consider the following example:

```
   import os
   import requests

   from oauthlib.oauth2 import LegacyApplicationClient
   from requests_oauthlib import OAuth2Session

   # Allows for no HTTPS verification. DO NOT USE IN PRODUCTION!
   os.environ['OAUTHLIB_INSECURE_TRANSPORT'] = '1'

   client_id = '2ec860db-ce81-49c2-b4b7-c71bcbee481f'
   client_secret = 'Test'
   username = 'admin'
   password = 'admin'

   oauth = OAuth2Session(client=LegacyApplicationClient(client_id=client_id))
   token = oauth.fetch_token(
       token_url='http://localhost:8060/oauth2/token',
       username=username,
       password=password,
       client_id=client_id,
       client_secret=client_secret,
       scope=['public-pricelist', 'pricelist-item', 'medical-medicament'],
   )
```

Above, the client requests the scopes `['public-pricelist', 'pricelist-item', 'medical-medicament']`. It is actually authorized for quite a bit more though, but that authorization is not required in the context of this specific integration. 

In the case of a mobile or web application, there would be a checkbox in the application itself disallowing certain scopes. The disallowance would have to be subsequently handled in the application itself as well, because it would be dependent on the data.

Given the above, I think this isn't necessary in the roadmap. Do you agree?